src/mpb.cpp

```diff
@@ -133,24 +135,34 @@ complex<double> eigenmode_amplitude(void *vedata, const vec &p,
   int nz = n[2];
   double r[3] = {0,0,0};
   vec p0(p - center);
-  LOOP_OVER_DIRECTIONS(p.dim, d)
-   r[d%3] = p0.in_direction(d) / s[d%3] + 0.5;
+  double phase = 0;
+  LOOP_OVER_DIRECTIONS(p.dim, d) {
+    double pd = p0.in_direction(d);
+    int i = d%3;
+    phase += edata->Gk[i] * pd; // k dot p
+    r[i] = pd / s[i] + 0.5;
+  }
   double rx = r[0], ry = r[1], rz = r[2];
 
   /* linearly interpolate the amplitude from MPB at point p */
   int x, y, z, x2, y2, z2;
   double dx, dy, dz;
 
   /* get the point corresponding to r in the epsilon array grid: */
-  x = pmod( int(rx * nx), nx );
-  y = pmod( int(ry * ny), ny );
-  z = pmod( int(rz * nz), nz );
+  x = int(rx * nx);
+  y = int(ry * ny);
+  z = int(rz * nz);
 
   /* get the difference between (x,y,z) and the actual point */
   dx = rx * nx - x;
   dy = ry * ny - y;
   dz = rz * nz - z;
 
+  /* wrap around to 0..n-1, assuming periodic boundaries */
+  x = pmod( x, nx );
+  y = pmod( y, ny );
+  z = pmod( z, nz );
```

Comment:
Don't these lines (162-164) conflict with line 170 further below in this file: `x = x % nx; y = y % ny; z = z % nz;`?